modules/Media.js

```diff
@@ -43,12 +44,23 @@ class Media extends React.Component {
     if (typeof query !== "string") query = json2mq(query);
 
     this.mediaQueryList = targetWindow.matchMedia(query);
+
+    const { onQueryStateChange } = this.props;
+    if (onQueryStateChange) {
+      this.mediaQueryList.addListener(onQueryStateChange);
```

Comment:
We already have a listener registered in `updateMatches`. We should call the `onChange` callback there instead of registering it as a separate listener.